third_party/amd/backend/compiler.py

```diff
@@ -116,6 +112,10 @@ def get_target_name(self, options) -> str:
         return f"hip:{options.arch}"
 
     def parse_options(self, opts) -> Any:
+        if "debug" in opts and opts["debug"]:
+            opts["enable_assertions"] = True
+            opts["sanitize_overflow"] = True
```

Comment:
Won't this break anyone who was passing `debug=True, sanitize_overflow=False`? I think you should check if `sanitize_overflow` is already in `opts`, for that matter probably `enable_assertions` as well.